src/components/myCoursesCard/courseDescription/LessonCourseSingle.jsx

```diff
@@ -1,11 +1,63 @@
+import { useState, useEffect } from 'react'
 import { useHistory } from 'react-router'
+import { useDispatch } from 'react-redux'
+import moment from 'moment'
 import { GET_COVERIMG } from '../../../utils/urlConstants'
+import { createLessonProgress } from '../../../actions/lessonProgressActions'
 
-const LessonCourseSingle = ({ data }) => {
+const LessonCourseSingle = ({
+  data,
+  userInfo,
+  creator,
+  isEnroll,
+  joinCourse,
+  enrolls,
+  clickLessonId,
+  setClickLessonId,
+  isCompleted
+}) => {
   const history = useHistory()
+  const dispatch = useDispatch()
+  const [isLessonCompleted, setIsLessonCompleted] = useState(false)
+  const [userId, setUserId] = useState('')
 
+  useEffect(() => {
+    setIsLessonCompleted(data?.lesson_progresses[0]?.isCompleted)
+    setUserId(data?.lesson_progresses[0]?.userId || null)
+  }, [data])
+
+  const lessonStart = async (id) => {
+    // console.log(id)
+    await setClickLessonId(id)
+    const addProgress = () => {
```

Comment:
seems like this is still WIP as addProgress is not being called so it does not seem to record or navigate to lesson